### PR TITLE
[Core][ModelpartIO] improve error message for hanging nodes

### PR DESCRIPTION
**Pull request by philbucher**
After speaking to @jcotela we realized that the error message is misleading, it is actually checking for hanging nodes, which is a no-go for the creation of the nodal graph.

Maybe we should add an option to skip the hanging nodes (separate discussion)

FYI @AndreasWinterstein

### CODE REVIEW

**Review by pooyan-dadvand — COMMENTED** [review]
Is not needed to be consecutive anymore?

**Comment by philbucher** [comment]
yes, but from how I read the code it is not checking for consecutivity but for hanging nodes
Not sure though, @jcotela ? :)

**Comment by jcotela** [comment]
@pooyan-dadvand the ModelPart still needs to be consecutive, but the ReorderModelPartIO can take care of that automatically. The problem here is that isolated nodes trigger the same check, but the message only refers to numbering, which can be confusing.

Honestly, I'd like to have a message that can suggest either as a cause for the error, but I do not know how to phrase it. From the point of view of the end user, the modelpart can be assumed to be consecutively numbered (the ImportModelPart python utility uses the ReorderModelPartIO by default).

**Comment by philbucher** [comment]
I improved the error message to reflect both possible sources of the error

**Comment by jcotela** [comment]
@philbucher I like the new error message much better. I was wondering: would it be helpful to also print the coordinates of the node? I am thinking of a scenario where a hanging node has been also renumbered, which should be unlikely, but one never knows...

**Comment by philbucher** [comment]
I wanted to do this, but I think at this time I don't have the coordinates available.
If you have an idea of how to get the coordinates then I am all ears :)

**Comment by jcotela** [comment]
@philbucher you are right, there is no easy way to get the coordinates from here.

File: kratos/sources/model_part_io.cpp [hunk 1]
```diff
@@ -571,17 +571,13 @@ std::size_t ModelPartIO::ReadNodalGraph(ConnectivitiesContainerType& rAuxConnect
...
+    for (const auto& r_conn : rAuxConnectivities) {
         n++;
-        if (inode->size() == 0)
-        {
-            std::stringstream msg;
-            msg << "Nodes are not consecutively numbered. Node " << n << " was not found in mdpa file." << std::endl;
-            KRATOS_ERROR << msg.str() << std::endl;
-        }
+        KRATOS_ERROR_IF(r_conn.size() == 0) << "Node #" << n << " caused an error during the construction of the nodal graph. Possible reasons are:\n"
+            << "The node is a hanging node node, not connected to any element or condition\n"
```
Comment (pooyan-dadvand) [inline kratos/sources/model_part_io.cpp]: "hanging node node"

**Comment by pooyan-dadvand** [comment]
Now I see, thanks for the explanation

**Review by jcotela — APPROVED** [review]

**Pull request merged**